src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts

```diff
@@ -72,13 +140,197 @@ describe('InstructorHomePageComponent', () => {
+      courseId: 'CS1231',
+      isArchived: true,
+    };
+
+    component.courseTabModels = activeCourseTabModels;
+    component.hasCoursesLoaded = true;
+    fixture.detectChanges();
+
+    expect(component.courseTabModels.length).toEqual(2);
+    expect(component.courseTabModels[0].course.courseId).toEqual('CS1231');
+    expect(component.courseTabModels[0].course.courseName).toEqual('Discrete Structures');
+
+    spyOn(simpleModalService, 'openConfirmationModal').and.callFake(() => {
+      return {
+        componentInstance: {
+          header: 'mock header', content: 'mock content', type: SimpleModalType.INFO,
+        },
+        result: Promise.resolve(),
+      };
+    });
+    spyOn(courseService, 'changeArchiveStatus').and.returnValue(of(courseArchive));
+
+    const courseButton: any = fixture.debugElement.nativeElement.querySelector('#btn-course');
+    courseButton.click();
+    const archiveButton: any = fixture.debugElement.nativeElement.querySelector('#btn-archive-course');
+    archiveButton.click();
+
+    expect(component.courseTabModels.length).toEqual(1);
+    expect(component.courseTabModels[0].course.courseId).toEqual('CS3281');
+    expect(component.courseTabModels[0].course.courseName).toEqual('Thematic Systems I');
+  });
+
+  it('should delete the entire course from the instructor', () => {
+    const courseToDelete: Course = {
+      courseId: 'CS1231',
+      courseName: 'Discrete Structures',
+      creationTimestamp: 1549095330000, // Saturday, 2 February 2019 16:15:30 GMT+08:00
+      deletionTimestamp: 0,
+      timeZone: 'Asia/Singapore',
+    };
+
+    component.courseTabModels = activeCourseTabModels;
+    component.hasCoursesLoaded = true;
+    fixture.detectChanges();
+
+    expect(component.courseTabModels.length).toEqual(2);
+    expect(component.courseTabModels[0].course.courseId).toEqual('CS1231');
+    expect(component.courseTabModels[0].course.courseName).toEqual('Discrete Structures');
+
+    spyOn(simpleModalService, 'openConfirmationModal').and.callFake(() => {
+      return {
+        componentInstance: {
+          header: 'mock header', content: 'mock content', type: SimpleModalType.WARNING,
+        },
+        result: Promise.resolve(),
+      };
+    });
+    spyOn(courseService, 'binCourse').and.returnValue(of(courseToDelete));
+
+    const courseButton: any = fixture.debugElement.nativeElement.querySelector('#btn-course');
+    courseButton.click();
+    const archiveButton: any = fixture.debugElement.nativeElement.querySelector('#btn-delete-course');
+    archiveButton.click();
+
+    expect(component.courseTabModels.length).toEqual(1);
+    expect(component.courseTabModels[0].course.courseId).toEqual('CS3281');
+    expect(component.courseTabModels[0].course.courseName).toEqual('Thematic Systems I');
+  });
+
+  it('should load courses of the current instructor', () => {
+    const activeCourses: Courses = {
+      courses: [
+        {
+          courseId: 'CS1231',
```

Comment:
What I intended in my earlier comment was for repeated data such as the course here (and in `activeCourseTabModels`) to be extracted into a single variable you can reuse. So you can define say for example `const testCourse1 = { ... }` and then use the object here as `courses: [ testCourse1, testCourse2 ]`. 

Thus you wouldn't have to unnecessarily repeat yourself and any future changes to test data can be centralised in the extracted variable.

Reply:
Thanks for the clarification! Overlooked on that.
Has updated accordingly.